### ISSUE TITLE
voting scheme

### ISSUE THREAD

**Issue by radu-diaconescu13** [VisualComputingInstitute/2D_lidar_person_detection#23]
Hello,

The DROW and DR-SPAAM papers contain information about some voting schemes when outputting the predictions.
I see there is a method in the utils.py file called group_predicted_center which is related to this voting, but it is not used anywhere.

Is there a reason for this? Did the experiments show that just using the NMS in the case of person detection is good enough?

Thank you

**Comment by Pandoro** [comment]
In the end it boils down to the function below it: https://github.com/VisualComputingInstitute/2D_lidar_person_detection/blob/99dd7a2a0d64252905e4f621e2c45be64b653a32/dr_spaam/dr_spaam/utils/utils.py#L608

The code is a bit convoluted, but if you check how the evaluation is done, eventually you end up at that function which is, as you say, pretty much just NMS. If I recall correctly, we initially started out with the scheme from DROWv2, which is represented by the function you referenced, but for speed and simplicity, simple NMS gave a better trade-off and we settled for that. The issue here was always that we, at that time, couldn't get the whole thing to work in an end-to-end differentiable and fast manner. As such it was always a somewhat finicky thing to put something non-learned on top of it that actually made the performance better without significant additional parameter tuning. I'd argue that today there are tons of new things one could try out in this domain and probably get a significantly better, fully end-to-end, trainable model. The big question is I guess whether you want to invest the time needed for that research, or you just want to use this model as is. In the latter case I'd recommend just sticking to NMS. If you do want to go for the group_predicted_center way. Make sure you actually have a look at what it does and do some automatic hyperparameter tuning on some validation set. We spend quite some CPU hours doing that in the past, and while intuitive values worked out just fine, tuned values did make a noticeable difference. On the other hand, this function is also quite a bit slower based on the fact that you need to the whole voting, blurring, and NMS scheme. I would expect that you could these days fairly easily create a torch version of the whole thing though, thus running quite a bit faster if speed is of interest.

I hope this helps, feel free to re-open if you have further questions about this.
